### PR TITLE
[DOC] some background reading for brands wanting to support their own spider(s)

### PR DESCRIPTION
**Pull request by mjoe999**
The few cannot do the job of the many! Each brand that looks after their own data is a step forward.

### CODE REVIEW

**Review by CloCkWeRX — APPROVED** [review]
I like it.

File: docs/MY_BRAND.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+shame as most brands invest a great deal of energy in getting
+it right on *their own website*.
+
+What if there was a way to take the effort invested in
+your website and in a lightweight manner provide it to others
+as you would wish it to be shown on digital maps? Change your data
+and watch it filter through into others maps and applications!
+There is now a frictionless way to do this. Simply [add a small piece
+of software](../README.md#contributing-code) to this project that talks to your website or brand
+marketing partner ([we interface to many of these](../locations/storefinders/README.md)).
```
Comment (CloCkWeRX) [inline docs/MY_BRAND.md:18]:
```suggestion
marketing partner ([we interface to many of these](../locations/storefinders/README.md)). Failing that, publish your store locations as https://schema.org/LocalBusiness data.
```

**Comment by davidhicks** [comment]
Good idea!

A few more suggestions:
- Suggest to brands that they update robots.txt to allow the ATP user agent string which includes "https://github.com/alltheplaces/alltheplaces" as an identifier, or be permissive in allowing everyone to access brand and location information pages.
- Suggest to brands that they disable geoblocking for brand/location information and cite a few reasons why, such as some of their customers may be living between or traveling between multiple countries and want to discover the brand locations before they arrive, and that geoblocking prevents the brand from being findable by users using common OSM-based mapping products.
- Perhaps note that ATP attempts to minimise impact to brand websites by using the minimal number of queries required (ideally just one for a statically served/cached JSON file of all location data), and having the official crawl performed once a week allowing ATP consumers to get fresh data once a week without every ATP consumer having to crawl brand websites themselves.
- If brands aren't yet ready with an easy-to-use source of location data such as a single statically served/cached JSON file of all location data, brands should implement HTTP 429 and robots.txt to advise crawlers such as ATP how to crawl location data should a legacy website be underperforming and is somehow impacted by a few thousand store information pages being crawled at the default ATP rate of 1 request/second.

File: locations/storefinders/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+## Store finders
+
+Many store locator pages are powered by back-end APIs served by a
+contracted third party _"brander marketing / store finder"_ company
+e.g. [RioSEO](https://www.rioseo.com/local-marketing-solutions/store-locator-software/).
+There is no need to duplicate the code to talk these store finder companies across
+multiple spiders. Rather a [common spider is written in this directory](./rio_seo.py)
+which can then be called on by brand specific code e.g.
+
+* [United Band](../spiders/united_bank_us.py#L5)
```
Comment (impiaaa) [inline locations/storefinders/README.md:11]: typo (band/bank)

File: docs/MY_BRAND.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+(in particular positions wrong by many metres) or missing entirely.
+The smaller a brand the less likely mapping companies will
+take an interest in *getting it right*. This is a
+shame as most brands invest a great deal of energy in getting
+it right on *their own website*.
+
+What if there was a way to take the effort invested in
+your website and in a lightweight manner provide it to others
+as you would wish it to be shown on digital maps? Change your data
+and watch it filter through into others maps and applications!
```
Comment (arrival-spring) [inline docs/MY_BRAND.md:15]:
```suggestion
and watch it filter through into other maps and applications!
```

File: locations/storefinders/README.md (hunk 2 above)
Comment (mjoe999) [reply locations/storefinders/README.md]: Thanks

**Comment by davidhicks** [comment]
> > Good idea!
> > A few more suggestions:
> > ```
> > * Suggest to brands that they update robots.txt to allow the ATP user agent string which includes "https://github.com/alltheplaces/alltheplaces" as an identifier, or be permissive in allowing everyone to access brand and location information pages.
> > 
> > * Suggest to brands that they disable geoblocking for brand/location information and cite a few reasons why, such as some of their customers may be living between or traveling between multiple countries and want to discover the brand locations before they arrive, and that geoblocking prevents the brand from being findable by users using common OSM-based mapping products.
> > 
> > * Perhaps note that ATP attempts to minimise impact to brand websites by using the minimal number of queries required (ideally just one for a statically served/cached JSON file of all location data), and having the official crawl performed once a week allowing ATP consumers to get fresh data once a week without every ATP consumer having to crawl brand websites themselves.
> > 
> > * If brands aren't yet ready with an easy-to-use source of location data such as a single statically served/cached JSON file of all location data, brands should implement HTTP 429 and robots.txt to advise crawlers such as ATP how to crawl location data should a legacy website be underperforming and is somehow impacted by a few thousand store information pages being crawled at the default ATP rate of 1 request/second.
> > ```
> 
> These are all good suggestions but verge more towards the technical and I was more aiming for a page that could be linked to for when chatting to the (technical) manager type people at a brand / company. They are not necessarily so savvy on the details and get scared of anything they might see as a project requiring infrastructure. The more techy stuff we fire their way the more they worry. A separate page along the lines of "what can you do to help us" could be on the cards?

Agreed with one exception that I think the geoblocking/robot blocking issue is widely misused and of utmost importance when discussing ATP with management of a brand/company. I'm no longer proposing that this document gets into the technical detail of robots.txt and HTTP 429 and the likes. Rather I think it'd be helpful in this management-focused document to suggest that management ensure the brand **does not** block location data because [ people planning to travel across borders | navigation apps in cars and phones | etc ] then can't discover brand locations. An example could include a tourist planning a trip and wanting to find a hotel near a tourist attraction that is walking distance to a laundromat and variety of restaurants. Geoblocking of hotel websites (probably to stop price discovery etc) almost always blocks ATP in the cross-fire. ATP just wants to get hotel locations and feature data which is something likely considered good for a hotel brand to have tourists discover in as many new ways as possible. ATP doesn't care if the brand blocks the rest of the site.

**Comment by JoeDixon-TomTom** [comment]
A few more words for @davidhicks .

**Pull request merged**